common/include/callback.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <functional>
+#include <iostream>
+
+class Callback {
+public:
+  virtual ~Callback() = default;
+  virtual void before_clk(){};
+  virtual bool after_clk() { return false; };
+};
+
+template <typename SC_TYPE> class Driver : public Callback {
+  using Writer = std::function<void(const SC_TYPE &)>;
+
+public:
+  Driver(CData &valid_, CData &ready_, Writer write_func_)
```

Comment:
`const CData &ready_` might be more accurate

Reply:
done